firedrake/interpolation.py

```diff
@@ -228,18 +238,18 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
     if not isinstance(expr, firedrake.Expression):
         if expr.ufl_domain() and expr.ufl_domain() != V.mesh():
             raise NotImplementedError("Interpolation onto another mesh not supported.")
-        ast, oriented, needs_cell_sizes, coefficients, _ = compile_expression_dual_evaluation(expr, to_element, coords,
-                                                                                              domain=V.mesh(), coffee=False)
-        kernel = op2.Kernel(ast, ast.name, requires_zeroed_output_arguments=True)
+        ast, oriented, needs_cell_sizes, coefficients, _ = compile_expression_dual_evaluation(expr, element, coords, coffee=False)
```

Comment:
The new codepath is quite a bit slower than the old one. This suggests that we need to bite the bullet and implement caching of these kernels just like we presently do for assembly kernels.

Reply:
Does the ability to store an `Interpolator` object not cut the mustard then?

Reply:
Not really.

Reply:
The general mantra is that the first time you run your code things might take a while, but new runs should be fast. This is now not the case for anything that does interpolation.